[brendanneojw] iP

brendanneojw opened the pull request:
### Siren ###

> "Simplicity is the ultimate sophistication." - Leonardo Di Vinci [(source)](https://www.modernfrugality.com/31-minimalism-quotes-that-perfectly-illustrate-less-is-more/)

Siren relieves you from the burden of recalling tasks, offering a:

- Text-based interface
- Simple learning curve
- _REMARKABLE_ speed and efficiency

Simply follow these steps:
1. Download it from [here](https://github.com/brendanneojw/ip)
2. Navigate to Launcher.java
3. Run it using Gradle
4. Start adding your tasks with the keywords "todo", "deadline" or "event" 😉

Features offered:

- [x] Managing tasks
- [x] Saving your tasks locally
- [x] GUI
- [ ] Reminders (coming soon)

Furthermore, it is **FREE FOR ALL TO USE!**

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

brendanneojw committed:
Implemented Level 0: Rename, Greet, Exit

brendanneojw committed:
Implemented Level 1: Echo

brendanneojw committed:
Implemented Level 2: Add, List

brendanneojw committed:
Implemented Level 3: Mark as Done

brendanneojw committed:
Implemented Level 4: ToDo, Event, Deadline
Changed if-else statements to switch statement

brendanneojw committed:
Implemented A-TextUiTesting

brendanneojw committed:
Implemented Level 5: Handle Errors
Made changes to follow Java coding standard

brendanneojw committed:
Implemented Level 6: Delete
Handled more invalid inputs for deadline and event actions

brendanneojw committed:
Implement Level 7: Save

brendanneojw committed:
Merge branch 'branch-Level-7'

brendanneojw committed:
Implement Level 8: Date and Time of various formats

brendanneojw committed:
Merge branch 'branch-Level-8'

brendanneojw committed:
Implement A-MoreOOP by adding extracting classes from Duke.java

brendanneojw committed:
Merge branch 'branch-A-MoreOOP'

brendanneojw committed:
Implement A-Packages by putting all classes into one package duke

brendanneojw committed:
Merge branch 'branch-A-Packages'

brendanneojw committed:
Merge branch 'add-gradle-support'

brendanneojw committed:
Implement A-Gradle

brendanneojw committed:
Implement A-JUnit tests on Task.java and Storage.java

brendanneojw committed:
Merge branch 'branch-A-JUnit'

brendanneojw committed:
Implement A.jar by adding details into build.gradle

brendanneojw committed:
Merge branch 'branch-A-Jar'

brendanneojw committed:
Implement A-JavaDoc into all class files

brendanneojw committed:
Merge branch 'branch-A-JavaDoc'

brendanneojw committed:
Implement A-CodingStandard by following Java coding standard

brendanneojw committed:
Merge branch 'branch-A-CodingStandard'

brendanneojw committed:
Implement Level 9: Find

brendanneojw committed:
Merge branch 'branch-Level-9'

brendanneojw committed:
Implement A-CheckStyle

brendanneojw committed:
Merge branch 'branch-A-CheckStyle'

craigtonlian commented on data/storageTestRewrite.txt:
```diff
@@ -0,0 +1 @@
+T |   | do well in CS2103T
```
Good goal to have 😄

craigtonlian commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,37 @@
+    public String lineToWriteFile() {
+        return "E | " + super.lineToWriteFile() + " | " + convertDateToString(this.from)
+                + " to " + convertDateToString(this.to);
+    }
```
Should this function be named as a verb, such as `convertToSaveFormat()`?

craigtonlian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,109 @@
+    private TaskList taskList;
+    private Storage storage;
+    private String userInput;
+    private boolean bye;
```
Maybe this can be named as `isBye` to reflect that it is a boolean?

craigtonlian commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,70 @@
+     */
+    public String lineToWriteFile() {
+        return (isDone ? "X" : " ") + " | " + this.description;
+    }
```
Maybe this can be renamed as a verb?

craigtonlian commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,148 @@
+        System.out.println("What can I do for you?");
+        horizontalLine();
+    }
+
```
Should this be named as a verb like `showWelcomeMessage()`?

craigtonlian commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,148 @@
+        System.out.println(taskList.taskArray.get(taskList.taskArray.size() - 1));
+        System.out.println("Now you have " + taskList.taskArray.size() + " tasks in the list.");
+    }
+
```
Some inconsistent tense for method names here could perhaps prove confusing. Maybe try to standardise it to present tense or the imperative mood? Like `deleteTaskOutput`, `addTaskOutput`, `markTask`, etc.

craigtonlian reviewed (commented):
Looks very nice! 😊

brendanneojw committed:
Test Launcher file with JavaFX

brendanneojw committed:
Implement Level 10: GUI with JavaFX

brendanneojw committed:
Merge branch 'branch-Level-10'

brendanneojw committed:
Implement A-Assertions

There is no usage of assertions in all classes.

The program is unable to check for different assumptions that can cause
runtime error.

Using assertions in different part of the programs allows us to test
for bug when dealing with different inputs that users can potentially
enter.

Java assertions is preferable because it can flag out assertions that
fails but will not terminate the program.

brendanneojw committed:
Merge pull request #1 from brendanneojw/branch-A-Assertions

Implement A-Assertions

brendanneojw committed:
Implement A-CodeQuality

Codes in some classes have borderline bad code quality.

Readability is not the priority at that point in time. As more features
are added, readability of the code will be beneficial to not only the
developers but to others reading it.

Refactor to improve code quality such as adhering to SLAP, shortening
long methods and structuring the code logically.

To achieve readability and understandability, these improvement of code
quality aspects will allow others to easily understand the codes.

brendanneojw committed:
Merge pull request #2 from brendanneojw/branch-A-CodeQuality

Implement A-CodeQuality

brendanneojw committed:
Implement A-CI

brendanneojw committed:
Merge branch 'branch-A-CI'

brendanneojw committed:
Implement B-Reminders on startup and adding remind command

brendanneojw committed:
Merge branch 'branch-BCD-Extension'

brendanneojw committed:
Fix an assertion in Storage.java that causes bug

brendanneojw committed:
Merge branch 'branch-A-Assertions'

brendanneojw committed:
Fix another assertion in Storage.java that causes bug

brendanneojw committed:
Merge branch 'branch-A-Assertions'

brendanneojw committed:
Fix B-Reminders implementation

brendanneojw committed:
Merge branch 'branch-BCD-Extension'

brendanneojw committed:
Implement A-Personality with new icons and new outputs

brendanneojw committed:
Merge branch 'branch-A-Personality'

brendanneojw committed:
Implement A-Personality with new title

brendanneojw committed:
Merge branch 'branch-A-Personality'

brendanneojw committed:
Implement A-UserGuide by updating README.md

brendanneojw committed:
Merge branch 'branch-A-UserGuide'

brendanneojw committed:
Fix UG rendering

brendanneojw committed:
Merge branch 'branch-A-UserGuide'

brendanneojw committed:
Fix more UG rendering issues

brendanneojw committed:
Merge branch 'branch-A-UserGuide'

brendanneojw committed:
Fix further UG rendering issues

brendanneojw committed:
Merge branch 'branch-A-UserGuide'

brendanneojw committed:
Finalise for iP

brendanneojw committed:
Fix JUnit testing file

brendanneojw committed:
Fix bugs for Reminder extension and improve code quality

brendanneojw committed:
Merge branch 'A-Release'
